go/chat/push.go

```diff
@@ -1314,21 +1314,20 @@ func (g *PushHandler) SubteamRename(ctx context.Context, m gregor.OutOfBandMessa
 }
 
 func (g *PushHandler) HandleOobm(ctx context.Context, obm gregor.OutOfBandMessage) (bool, error) {
+	// Don't process messages if we have not started.
+	g.Lock()
```

Comment:
Why does it hold this lock the whole function call?

Reply:
holds the lock while we register the function with the error group

Reply:
I think this is dangerous, what is this actually solving for?

Reply:
All of these calls just decode and kick off a thread though, what is the concern? The fix isjust to allow concurrent test runs @AMarcedone was trying to track down a race. Otherwise the tests can panic with a log after completion

Reply:
They also call `WaitForTurn`

Reply:
Ah but I guess that guy just returns a channel.

Reply:
Also, I've never seen a log after completion from this place, how is his test triggering it?

Reply:
Log after completion happens because we're handling some OOBM after shutdown, log could happen anywhere during processing inboxsource, convsource etc

Reply:
I dislike changing anything just to serve tests, let me look at it a little more.

Reply:
ok

Reply:
@mmaxim bump on this